src/lib/coda_networking/coda_networking.ml

```diff
@@ -661,13 +663,19 @@ module Make (Inputs : Inputs_intf) = struct
     | Ok (Some data) ->
         return (Ok data)
     | Ok None ->
-        Logger.faulty_peer t.logger ~module_:__MODULE__ ~location:__LOC__
-          !"get no response from %{sexp: Peer.t}"
-          peer ;
+        let%bind () =
+          Trust_system.(
+            record t.trust_system t.logger peer.host
+              Actions.
+                ( Violated_protocol
+                , Some
+                    ( "When querying preferred peer $peer, got no response"
+                    , [("peer", Peer.to_yojson peer)] ) ))
+        in
         let peers = get_random_peers () in
         try_non_preferred_peers t envelope peers ~rpc
     | Error _ ->
-        Logger.faulty_peer t.logger ~module_:__MODULE__ ~location:__LOC__
+        Logger.error t.logger ~module_:__MODULE__ ~location:__LOC__
```

Comment:
Looking at `query_peer`, it calls `try_call_rpc` which calls into Jane Street RPC and passes the connection object to a callback, which is `rpc` here. We need to figure out what punishments apply there. If the connection times out it should be a soft punishment, if RPC handshaking fails they should be banned, and errors that come from the callback probably need case-by-case logic. That work can be a separate PR though